src/layer/poi.js

```diff
@@ -0,0 +1,103 @@
+import * as label from "../constants/label.js";
+import * as Color from "../constants/color.js";
+
+var iconDefs = {
+  bar: ["bar", "beer"],
+  hospital: "hospital",
+  medical: ["doctors", "clinic"],
+  parking: "parking",
+};
+
+export const poi = {
+  id: "poi",
+  type: "symbol",
+  paint: {
+    "text-halo-color": Color.backgroundFill,
+    "text-halo-width": 1.5,
+    "icon-halo-width": 0.4,
+    "text-halo-blur": 1,
+    "icon-halo-blur": 0.2,
+    "text-color": [
+      "match",
+      ["get", "class"],
+      ["bar", "beer"],
+      Color.poi.consumer,
+      ["hospital", "parking"],
+      Color.poi.infrastructure,
+      Color.poi.infrastructure,
+    ],
+  },
+  filter: [
+    ">=",
+    ["zoom"],
+    [
+      "match",
+      ["get", "subclass"],
+      "hospital",
+      15,
+      ["bar", "beer"],
+      16,
+      ["clinic", "doctors", "parking"],
+      17,
+      99,
+    ],
+  ],
+  layout: {
+    "text-font": ["Metropolis Light"],
+    "icon-optional": false,
+    "text-size": {
+      base: 1.0,
+      stops: [
+        [15, 10],
+        [17, 12],
+      ],
+    },
+    "icon-image": [
+      "match",
+      ["get", "subclass"],
+      iconDefs.bar,
+      "poi_martini_glass",
+      iconDefs.medical,
+      "poi_health_cross",
+      iconDefs.hospital,
+      "poi_hospital",
+      iconDefs.parking,
+      "poi_p",
+      "poi_square_dot", //icon for generic POI, not currently used
+    ],
+    "icon-size": 1.0,
+    "text-field": label.localizedName,
+    "text-variable-anchor": ["left", "right", "bottom"],
+    "text-justify": "auto",
+    "text-radial-offset": 1.2,
+    "text-max-width": 5,
+    "icon-padding": 0,
+    "text-padding": 0,
+    "icon-allow-overlap": false,
+  },
+  source: "openmaptiles",
+  "source-layer": "poi",
+};
+
+export const legendEntries = [
+  {
+    description: "Bar or Pub",
+    layers: [poi.id],
+    filter: ["in", ["get", "subclass"], ["literal", iconDefs.bar]],
+  },
```

Comment:
I think this section should be ordered from rarest / largest service area to most common / smallest service area. Community anchor institutions (#435) would end up first, while “Bar or pub” and other storefront POIs would end up last.

Reply:
Resolved in 107e5be.